WordPress/src/main/java/org/wordpress/android/models/ReaderBlog.java

```diff
@@ -26,28 +25,6 @@
     private String mImageUrl;
     private String mFeedUrl;
 
-    public static ReaderBlog fromRecommendedBlogJson(@NotNull JSONObject json) {
-        ReaderBlog blog = new ReaderBlog();
-        blog.blogId = json.optLong("blog_id");
-        blog.feedId = json.optLong("feed_id");
-        blog.setName(JSONUtils.getStringDecoded(json, "name"));
-        blog.setUrl(JSONUtils.getString(json, "url"));
-        blog.setDescription(JSONUtils.getStringDecoded(json, "description"));
-        blog.setFeedUrl(JSONUtils.getString(json, "feed_url"));
-        blog.isJetpack = JSONUtils.getBool(json, "is_jetpack");
-        blog.isPrivate = JSONUtils.getBool(json, "is_private");
-        blog.numSubscribers = json.optInt("subscribers_count");
-        JSONObject jsonNotification = JSONUtils.getJSONChild(json, "subscription/delivery_methods/notification");
-        if (jsonNotification != null) {
-            blog.isNotificationsEnabled = JSONUtils.getBool(jsonNotification, "send_posts");
-        }
-        JSONObject jsonIcon = json.optJSONObject("icon");
-        if (jsonIcon != null) {
-            blog.setImageUrl(JSONUtils.getString(jsonIcon, "img"));
-        }
```

Comment:
@AdamGrzybkowski now that we've removed this parsing method, I think we can move this icon parsing logic to the existing block.

Reply:
hmm I'm not sure I follow, there is a logic already to parse the `icon` in the `fromJson()` method 🤔

Reply:
oh yes! It exists outside else block. I noticed icon was not shown for a recommended site but it was on site details screen. I'll see if I can reproduce it.

Reply:
![icon](https://user-images.githubusercontent.com/1405144/94528712-66331980-0256-11eb-94c5-ac3e34ac2f37.gif)

Reply:
hmm could you check the logs if there's something? 
I will try to reproduce it myself

Reply:
BlogId, FeedId:
`{"ID":160909219,"feed_ID":97888741}`

Logs:

```
W/Glide: Load failed for https://startupandroid.com/wp-content/uploads/2019/11/cropped-19ccf390-b650-40ca-8ad8-994e509860c8.jpg with size [110x110]
    class com.bumptech.glide.load.engine.GlideException: Failed to load resource
    There was 1 cause:
    com.android.volley.NoConnectionError(java.net.UnknownHostException: Unable to resolve host "startupandroid.com": No address associated with hostname)
     call GlideException#logRootCauses(String) for more detail
      Cause (1 of 1): class com.bumptech.glide.load.engine.GlideException: Fetching data failed, class java.io.InputStream, REMOTE
    There was 1 cause:
    com.android.volley.NoConnectionError(java.net.UnknownHostException: Unable to resolve host "startupandroid.com": No address associated with hostname)
     call GlideException#logRootCauses(String) for more detail
        Cause (1 of 1): class com.bumptech.glide.load.engine.GlideException: Fetch failed
```

Seems like it was a jetpack site and `https://startupandroid.com` wasn't reachable 🤔.

Reply:
> Seems like it was a jetpack site

What does this mean exactly?

Why could this be loaded on the Site details screen if it wasn't reachable? 🤔

Reply:
Site details response includes image url in a different format

https://i1.wp.com/startupandroidcom.wpcomstaging.com/wp-content/uploads/2019/11/cropped-19ccf390-b650-40ca-8ad8-994e509860c8.jpg?fit=512%2C512&ssl=1

<img width="1332" alt="Screenshot 2020-09-29 at 1 41 23 PM" src="https://user-images.githubusercontent.com/1405144/94531026-8f08de00-0259-11eb-8d62-971ac7c3896f.png">

Not sure if icon url included for recommended site should be formatted in some manner. Looks like it is a question for @bluefuton .

Reply:
interesting 🤔 
thanks @ashiagr for spotting this!

Reply:
Looks like there's a base url difference
https://startupandroidcom.wpcomstaging.com/
vs
https://startupandroidcom.com/

First one is reachable, second one is not.

Reply:
yeah... I guess we have to wait for @bluefuton answer, as I don't think we can do much here

Reply:
I looked into this. It's a Jetpack site and `/sites/:site` endpoint contacts the site directly for these details (including the icon). The `/read/site/:site` relies on the information in the shadow/cache blog. I've tried a full sync but that hasn't made a difference.

The background is: this Atomic site hasn't launched yet and the domain is inactive. I don't think it should have been recommended, so I've created a task for us to filter out sites with *.wpcomstaging.com domains.

Reply:
Yes we shouldn't recommend such sites. Thank you for this update 🙇‍♀️.